consensus/types/state.go

```diff
@@ -77,10 +77,20 @@ type RoundState struct {
 // RoundStateEvent returns the H/R/S of the RoundState as an event.
 func (rs *RoundState) RoundStateEvent() types.EventDataRoundState {
 	edrs := types.EventDataRoundState{
-		Height:     rs.Height,
-		Round:      rs.Round,
-		Step:       rs.Step.String(),
-		RoundState: rs,
+		Height: rs.Height,
+		Round:  rs.Round,
+		Step:   rs.Step.String(),
+		// send only fields needed by makeRoundStepMessages
+		RoundState: &RoundState{
```

Comment:
we need to have a larger discussion/investigation about the races this is meant to address. AFAIK they only started showing up after a seemingly innocuous change to go-wire, which has been rolled back, so we shouldn't need to do this now. But we will need to investigate and solve it properly soon ...